drivers/sensor/mpu9250/mpu9250.c

```diff
@@ -0,0 +1,299 @@
+	drv_data->magn_z = sys_be16_to_cpu(buf[9]);
+#endif
+
+	return 0;
+}
+
+static const struct sensor_driver_api mpu9250_driver_api = {
+#if CONFIG_MPU9250_TRIGGER
+	.trigger_set = mpu9250_trigger_set,
+#endif
+	.sample_fetch = mpu9250_sample_fetch,
+	.channel_get = mpu9250_channel_get,
+};
+
+int mpu9250_init(const struct device *dev)
+{
+	struct mpu9250_data *drv_data = dev->data;
+	const struct mpu9250_config *cfg = dev->config;
+	uint8_t id, i;
+
+	drv_data->i2c = device_get_binding(cfg->i2c_label);
+	if (drv_data->i2c == NULL) {
+		LOG_ERR("Failed to get pointer to %s device",
+			    cfg->i2c_label);
+		return -EINVAL;
+	}
+
+	/* check chip ID */
+	if (i2c_reg_read_byte(drv_data->i2c, cfg->i2c_addr,
+			      MPU9250_REG_CHIP_ID, &id)) {
+		LOG_ERR("Failed to read chip ID.");
+		return -EIO;
+	}
+
+	if (id != MPU9250_CHIP_ID && id != MPU9250_CHIP_ID) {
+		LOG_ERR("Invalid chip ID.");
+		return -EINVAL;
+	}
+
+	/* wake up chip */
+	if (i2c_reg_update_byte(drv_data->i2c, cfg->i2c_addr,
+				MPU9250_REG_PWR_MGMT1, MPU9250_SLEEP_EN, 0)) {
+		LOG_ERR("Failed to wake up chip.");
+		return -EIO;
+	}
+
+	/* set accelerometer full-scale range */
+	for (i = 0U; i < 4; i++) {
+		if (BIT(i+1) == CONFIG_MPU9250_ACCEL_FS) {
+			break;
+		}
+	}
+
+	if (i == 4U) {
+		LOG_ERR("Invalid value for accel full-scale range.");
+		return -EINVAL;
+	}
+
+	if (i2c_reg_write_byte(drv_data->i2c, cfg->i2c_addr,
+			       MPU9250_REG_ACCEL_CFG,
+			       i << MPU9250_ACCEL_FS_SHIFT)) {
+		LOG_ERR("Failed to write accel full-scale range.");
+		return -EIO;
+	}
+
+	drv_data->accel_sensitivity_shift = 14 - i;
+
+	/* set gyroscope full-scale range */
+	for (i = 0U; i < 4; i++) {
+		if (BIT(i) * 250 == CONFIG_MPU9250_GYRO_FS) {
+			break;
+		}
+	}
+
+	if (i == 4U) {
+		LOG_ERR("Invalid value for gyro full-scale range.");
+		return -EINVAL;
+	}
+
+	if (i2c_reg_write_byte(drv_data->i2c, cfg->i2c_addr,
+			       MPU9250_REG_GYRO_CFG,
+			       i << MPU9250_GYRO_FS_SHIFT)) {
+		LOG_ERR("Failed to write gyro full-scale range.");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(drv_data->i2c, cfg->i2c_addr,
+			       MPU9250_REG_CONFIG,
+			       CONFIG_MPU9250_GYRO_DLPF & MPU9250_DLPF_MASK)) {
+		LOG_ERR("Failed to write gyro digital LPF settings.");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(drv_data->i2c, cfg->i2c_addr,
+			       MPU9250_REG_SR_DIV,
+			       CONFIG_MPU9250_GYRO_ODR)) {
+		LOG_ERR("Failed to write gyro ODR divider.");
+		return -EIO;
+	}
+
+	drv_data->gyro_sensitivity_x10 = mpu9250_gyro_sensitivity_x10[i];
+
+#ifdef CONFIG_MPU9250_TRIGGER
+	if (mpu9250_init_interrupt(dev)) {
+		LOG_DBG("Failed to initialize interrupts.");
+		return -EIO;
+	}
+#endif
+
+#ifdef CONFIG_MPU9250_MAGN_EN
+	if (ak8963_init(dev)) {
+		return -EIO;
+	}
+#endif
+
+	return 0;
+}
+
+static struct mpu9250_data mpu9250_driver;
+static const struct mpu9250_config mpu9250_cfg = {
+	.i2c_label = DT_INST_BUS_LABEL(0),
+	.i2c_addr = DT_INST_REG_ADDR(0),
```

Comment:
please update driver to use `i2c_dt_spec` (see other drivers for reference, many have been ported)